backend/src/routes/transactions.py

```
@@ -188,3 +204,99 @@ def get_all_transactions(
+                # get the transfer
+                stmt = select(Transfer).where(Transfer.transaction_id == transaction_id)
+                transfer = session.exec(stmt).one()
+                assert transfer
+                return {"transaction": transaction, "transfer": transfer}
+
+    except Exception as e:
+        logger.exception(e)
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="An error occurred while fetching transaction",
+        )
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)